### PR TITLE
Search bar

### LINKED ISSUE #1: Workshop Search
Add a search bar above the workshop page, and return any workshop with a string that matches the search.

Marc says this can happen in 12 lines of code.

### PR DESCRIPTION
**Pull request by philipqueen**
Adds a search bar to the `Schedule`, `Favorites`, and `Teachers` pages. Fixes #1.

On schedule and favorites, it searches through the workshop title, type, description, location, and primary and coteachers. On the teachers page, it searches through their full name. You can clear the search by clicking on the x.

I haven't tested it on Android.

Screen Shot 2024-09-10 at 9 10 35 AM
Screen Shot 2024-09-10 at 9 10 15 AM

### CODE REVIEW

File: src/components/SearchBar.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { StyleSheet, Text, TouchableOpacity, View, TextInput } from 'react-native';
+import { RFValue } from "react-native-responsive-fontsize";
+
+export default class SearchBar extends React.Component {
+    constructor(props) {
+        super(props);
+      }
+    render(){
```
Comment (mlashner) [inline src/components/SearchBar.js]: nit: spacing
```suggestion
        super(props);
    }
    
    render() { 
```

File: src/components/SearchBar.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        flexDirection: 'row',
+        justifyContent: 'center',
+        alignItems: 'center',
+    },
+    clearButton: {
+        fontSize: RFValue(16),
+        color: '#5d8da0',
+        fontFamily: 'helvetica77',
+    },
+});
```
Comment (mlashner) [inline src/components/SearchBar.js]: nit: newline
```suggestion
});

```

File: src/screens/Teachers.js [hunk 3]
```diff
@@ -12,10 +13,30 @@ export default class TeachersScreen extends React.Component {
...
             data: props.route.params.teachers ?? [],
-            user: props.route.params.user ?? {}
+            user: props.route.params.user ?? {},
+            searchQuery: '',
         }
+        this.clearSearch = this.clearSearch.bind(this);
       }
 
+    handleSearch = (text) => {
+        this.setState({searchQuery: text})
```
Comment (mlashner) [inline src/screens/Teachers.js]: nit: spacing
```suggestion
        this.setState({ searchQuery: text })

```

File: src/screens/Teachers.js [hunk 4]
```diff
@@ -12,10 +13,30 @@ export default class TeachersScreen extends React.Component {
...
+    }
+
+    clearSearch() {
+        this.setState({searchQuery: ''})
```
Comment (mlashner) [inline src/screens/Teachers.js]: nit: spacing
```suggestion
        this.setState({ searchQuery: '' })

```

File: src/screens/Teachers.js [hunk 5]
```diff
@@ -12,10 +13,30 @@ export default class TeachersScreen extends React.Component {
...
+    }
+
+    getFilteredData() {
+        const {data, searchQuery} = this.state;
+        const sortedData = data.sort((prev, next)=>prev.fullName.localeCompare(next.fullName))
```
Comment (mlashner) [inline src/screens/Teachers.js]: nit: spacing
```suggestion
        const sortedData = data.sort((prev, next) => prev.fullName.localeCompare(next.fullName))

```

File: src/screens/Teachers.js [hunk 6]
```diff
@@ -12,10 +13,30 @@ export default class TeachersScreen extends React.Component {
...
+    handleSearch = (text) => {
+        this.setState({searchQuery: text})
+    }
+
+    clearSearch() {
+        this.setState({searchQuery: ''})
+    }
+
+    getFilteredData() {
+        const {data, searchQuery} = this.state;
```
Comment (mlashner) [inline src/screens/Teachers.js]: nit: spacing
```suggestion
        const { data, searchQuery } = this.state;

```

File: src/screens/Teachers.js [hunk 7]
```diff
@@ -12,10 +13,30 @@ export default class TeachersScreen extends React.Component {
...
             isLoading: true,
             data: props.route.params.teachers ?? [],
-            user: props.route.params.user ?? {}
+            user: props.route.params.user ?? {},
+            searchQuery: '',
         }
+        this.clearSearch = this.clearSearch.bind(this);
       }
 
+    handleSearch = (text) => {
```
Comment (mlashner) [inline src/screens/Teachers.js:22]: You don't need to bind this function?

File: src/screens/Teachers.js [hunk 8]
```diff
@@ -12,10 +13,30 @@ export default class TeachersScreen extends React.Component {
...
+        this.setState({searchQuery: text})
+    }
+
+    clearSearch() {
+        this.setState({searchQuery: ''})
+    }
+
+    getFilteredData() {
```
Comment (mlashner) [inline src/screens/Teachers.js:32]: You don't need to bind this function?

**Review by mlashner — APPROVED** [review]
LGTM if it's all tested and working! Curious why we needed to bind the clear function but not the others?

Some little style nits just personal preference for readability

File: src/screens/Teachers.js (hunk 8 above)
Comment (philipqueen) [reply src/screens/Teachers.js:32]: `getFilteredData()` is only called in this class's render method, so it should be good without the bind

File: src/screens/Teachers.js (hunk 7 above)
Comment (philipqueen) [reply src/screens/Teachers.js:22]: I'm really unclear *why* this doesn't need to be binded, but it does seem to function properly without the bind.

Comment (philipqueen) [reply src/screens/Teachers.js:22]: Digged a little deeper, and the arrow function syntax does the same thing as binding. So we have two different syntaxes that achieve the same thing here.

**Pull request merged**